# Visualisierung: Offene Lehrstellen LENA

## Libraries importieren

In [1]:
import pandas as pd

## Daten einlesen

Die Daten sind als csv-Daten im gleichen Ordner abgelegt, versehen mit dem Datum wann die Seite gescrapt wurde.

In [2]:
path = "anzahl_jobs_04-11-2018_NEU.csv"

with open(path) as f:
    dateiinhalt = f.read()

In [3]:
df = pd.read_csv(path)

In [4]:
df.head(3)

Beruf  Anzahl
0             Abdichter/in EFZ     135
1  Abdichtungspraktiker/in EBA      44
2        Agrarpraktiker/in EBA      57

Spalten und Reihen sind vertauscht. Die csv-Datei muss transponiert und (ACHTUNG WICHTIG) **neu abgespeichert** werden.

In [7]:
df.dtypes

Beruf     object
Anzahl     int64
dtype: object

In [10]:
df.describe()

Anzahl
count   241.000000
mean    167.261411
std     413.445273
min       0.000000
25%      10.000000
50%      40.000000
75%     125.000000
max    4622.000000

**RESULTAT 1:** Im Lehrstellennachweis LENA gibt es **241** verschiedene Berufe.

In [8]:
df['Anzahl'].sum()

40310

**RESULTAT 2:** Im Lehrstellennachweis LENA gab es am 4.November 2018 **40310** offene Lehrstellen.

In [14]:
df.groupby('Beruf')['Anzahl'].mean().sort_values(ascending=False).head(20)

Beruf
Kaufmann/-frau EFZ, erw. Grundbildung                           4622
Detailhandelsfachmann/-frau EFZ, Schwerpunkt Beratung           2116
Fachmann/-frau Gesundheit EFZ                                   1903
Elektroinstallateur/in EFZ                                      1534
Detailhandelsfachmann/-frau EFZ, Schwerpunkt Bewirtschaftung    1463
Koch/Köchin EFZ                                                 1142
Logistiker/in EFZ                                               1102
Polymechaniker/in EFZ                                           1084
Detailhandelsassistent/in EBA                                    963
Informatiker/in EFZ                                              836
Fachmann/-frau Betreuung EFZ                                     787
Coiffeur/-euse EFZ                                               775
Montage-Elektriker/in EFZ                                        718
Sanitärinstallateur/in EFZ                                       662
Kaufmann/-frau EFZ, Basis-Gr

**RESULTAT 3:** Aktuell hat es mit Abstand die meisten offenen Stellen im Bereich Kaufmänner/-frauen (kurz KV).

In [15]:
df.groupby('Beruf')['Anzahl'].mean().sort_values(ascending=False).tail(20)

Beruf
Keramiker/in EFZ                                           1
Vergolder/in-Einrahmer/in EFZ                              1
Polisseur/-euse EBA                                        1
Gewebegestalter/in EFZ                                     1
Korb- und Flechtwerkgestalter/in EFZ                       1
Holzhandwerker/in EFZ                                      1
Büchsenmacher/in EFZ                                       1
Forstpraktiker/in EBA                                      1
Formenpraktiker/in EBA                                     1
Marmorist/in EFZ                                           1
Bühnentänzer/in EFZ                                        0
Gästeempfangsfachmann/-frau HGA EFZ (Kt. Verordnung VS)    0
Geigenbauer/in EFZ                                         0
Steinsetzer/in EBA                                         0
Architekturmodellbauer/in EFZ                              0
Theatermaler/in EFZ                                        0
Messerschmied/in E

**RESULTAT 4:** Es gibt Ausbildungen, davon wird keine einzige in der Schweiz aktuell angeboten: wie zB Steinsetzer, Glasmaler oder Küfer.